miden-tx/src/tests/kernel_tests/mod.rs

```diff
@@ -29,77 +29,56 @@ pub fn read_root_mem_value<H: Host>(process: &Process<H>, addr: u32) -> Word {
     process.get_mem_value(ContextId::root(), addr).unwrap()
 }
 
+/// Given an slice of [Note]s, generates a procudere to initialize the VM's memory with the note's
+/// data.
 pub fn output_notes_data_procedure(notes: &[Note]) -> String {
```

Comment:
This should allow one to simplify the context builder, and remove notes which are no longer needed, since the number of notes now can vary.

In the future it should also support notes with a variable number of assets.